src/Microsoft.Azure.SignalR/HubHost/ServiceLifetimeManagerBase.cs

```diff
@@ -253,6 +255,28 @@ protected static bool IsInvalidArgument(IReadOnlyList<object> list)
             {
                 await task(message);
             }
+            catch (TimeoutException ex)
+            {
+                string connectionId = null;
+                switch (message)
+                {
+                    case JoinGroupWithAckMessage jmg:
```

Comment:
I am concerned about the extensibility of this change. How about adding an interface `IConnectionAckableMessage : IAckableMessage`?

Reply:
Yes. Also an option considered which will include protocol change. Do we need any further discuss?